docs/data/base/components/form-control/form-control.md

```diff
@@ -0,0 +1,58 @@
+---
+product: base
+title: React form control
+components: FormControlUnstyled
+packageName: '@mui/base'
+---
+
+# Unstyled form control
+
+<p class="description">
+  The FormControlUnstyled is a utility that lets you associate a form input with auxillary components,
+  such as labels, error indicators or helper text.
+</p>
+
+{{"component": "modules/components/ComponentLinkHeader.js", "design": false}}
+
+## FormControlUnstyled
+
+FormControlUnstyled wraps an input and other components and enables reflecting the input's state in these other components.
+
+For instance, you may show additional element asking the user to enter a value if the input is empty or display a warning icon when the entered value is incorrect.
+
+The FormControlUnstyled provides a context that can be read by the useFormControl hook.
+
+## useFormControl hook
+
+The useFormControl hook can be used to enable integration between custom form inputs and FormControlUnstyled.
+Additionally, you can read the form control's state and react to its changes in a custom component.
+
+The demo below shows both.
+`CustomInput` is a wrapper around a native HTML `input` that adds FormControlUnstyled integration.
+`ControlStateDisplay` reads the state of the form control and displays it as text.
+
+{{"demo": "UseFormControl.js", "defaultCodeOpen": false}}
```

Comment:
Suggestion: Since we have only 1 demo in this page, we can show the code by default.

Reply:
I decided not to show it as the code in the preview does not explain the concept. It's the code outside the main function that matters in this case.